snuba/datasets/errors_replacer.py

```diff
@@ -220,14 +246,31 @@ def get_projects_query_flags(
             exclude_groups_key, float("inf"), now - settings.REPLACER_KEY_TTL
         )
 
+    for type_key in exclude_groups_keys_replacement_types:
+        p.smembers(type_key)
+
+    # 'results' list looks like:
+    # [needs_final..., exclude_groups..., replacement_types...]
     results = p.execute()
 
     needs_final = any(results[: len(s_project_ids)])
+
+    exclude_groups_results = results[
+        len(s_project_ids) + 1 : -len(exclude_groups_keys_replacement_types)
```

Comment:
Are you sure you do not have an off by one error here? shouldn't this slice start at `len(s_project_ids)` instead? Or am I missing something ?

Reply:
This is how it was originally done, in master you'll see that `needs_final = any(results[: len(s_project_ids)])` and `exclude_groups` is indexing `results` like so: `results[(len(s_project_ids) + 1) :: 2]`. I believe this is because the excludes_groups slice of results is in the form `[int, list, int, list..]` and we just want to pick up the lists, and the off by one is just skipping the first int in the list.

https://github.com/getsentry/snuba/blob/bfc46623fdb42091791da3c4921100e1d84df119/snuba/datasets/errors_replacer.py#L225-L228